### ISSUE TITLE
[BUG] Shadow Token Mask 逻辑可能反转

### ISSUE THREAD

**Issue by huthvincent** [huthvincent/KV-Cache-Compression-in-Reinforcement-Learning#2]
**Labels**: `bug`, `critical`, `loss-function`

### 问题描述
`SMD/src/shadow_distillation_loss.py` 中 `visibility_ratio < 1.0` 的判断逻辑可能与预期相反。

### 当前代码
```python
# 文件: SMD/src/shadow_distillation_loss.py
# 行号: 约 130-140 行

visibility_ratio = resp_mask.float().sum(dim=-1) / total_len
shadow_token_masks.append(
    (visibility_ratio < 1.0).to(logits.device)  # 可见度 < 100% 的 token
)
```

### 问题分析
- 注释说"如果 visibility_ratio < 1.0, 说明有 KV 被压缩，token 在稀疏条件下生成"
- 但这意味着**所有**在压缩后生成的 token 都被标记为"可信"
- 实际上应该是：只有那些**完全**在保留的 KV 上下文中生成的 token 才是"on-policy faithful"

### 建议修复
```python
# 修复：检查每个 token 是否只依赖于保留的 KV 位置
retention_ratio = getattr(args, "shadow_retention_ratio", 0.5)
# 只有当可见度接近 retention_ratio 时，token 才是在压缩条件下生成的
shadow_token_masks.append(
    (visibility_ratio <= retention_ratio + 0.1).to(logits.device)
)
```

### 影响范围
- 影响 shadow mask 的正确性
- 可能导致错误的 token 被标记为"可信"
